classes/core/PKPRequest.inc.php

```diff
@@ -154,24 +151,26 @@ function getIfModifiedSince() {
 
 	/**
 	 * Get the base URL of the request (excluding script).
+	 * @param $allowProtocolRelative boolean True iff protocol-relative URLs are allowed
 	 * @return string
 	 */
-	function getBaseUrl() {
+	function getBaseUrl($allowProtocolRelative = false) {
 		$_this =& PKPRequest::_checkThis();
 
-		if (!isset($_this->_baseUrl)) {
```

Comment:
Yes, that's correct. Practically speaking I don't think re-executing this code every time (as opposed to caching) will have a performance impact.